app/controllers/provider/admin/applications_controller.rb

```diff
@@ -103,7 +103,7 @@ def destroy
       redirect_to provider_admin_applications_path
     else
       flash[:notice] = 'Not possible to delete application'
-      redirect_to :back
+      redirect_back(fallback_location: root_path)
```

Comment:
Maybe we want to redirect to the list of applications `provider_admin_applications_path`? Or to the edit page of the application. @3scale/system-ux  what is your thought about it?